parser_library/src/workspaces/workspace.h

```diff
@@ -204,6 +204,28 @@ class workspace : public diagnosable_impl
     std::vector<const processor_file_compoments*> find_related_opencodes(const resource_location& document_loc) const;
     void filter_and_close_dependencies(std::set<resource_location> files_to_close_candidates,
         const processor_file_compoments* file_to_ignore = nullptr);
+
+    template<typename WS_INSTANCE>
+    static auto apply_options_to(WS_INSTANCE& ws_instance, const resource_location& pgm_location, asm_option& opts)
+    {
+        using PROC_GROUP_T = decltype(ws_instance.m_configuration.get_proc_grp_by_program(
+            program { resource_location(), std::nullopt, config::assembler_options {} }));
+
+        std::pair<const program*, PROC_GROUP_T> pgm_proc_grp_pair;
+        auto& [pgm, proc_grp] = pgm_proc_grp_pair;
+
+        if (pgm = ws_instance.m_configuration.get_program(pgm_location); !pgm)
+            ws_instance.implicit_proc_grp.apply_options_to(opts);
+        else
+        {
+            if (proc_grp = ws_instance.m_configuration.get_proc_grp_by_program(*pgm); proc_grp)
+                proc_grp->apply_options_to(opts);
+
+            pgm->asm_opts.apply_options_to(opts);
+        }
+
+        return pgm_proc_grp_pair;
+    }
```

Comment:
I had something like the following in mind...
```suggestion
    std::pair<const program*,const processor_group*> apply_options_to(const resource_location& pgm_location, asm_option& opts)
    {
        std::pair<const program*,const processor_group*> result = {};
        auto& [pgm, proc_grp] = result;

        if (pgm = m_configuration.get_program(pgm_location); !pgm)
            implicit_proc_grp.apply_options_to(opts);
        else
        {
            if (proc_grp = m_configuration.get_proc_grp_by_program(*pgm); proc_grp)
                proc_grp->apply_options_to(opts);

            pgm->asm_opts.apply_options_to(opts);
        }

        return result;
    }
```

Reply:
Then I think it will fail to compile [here](https://github.com/BroadcomMFD-DevOps/che-che4z-lsp-for-hlasm/blob/43b8bf87d4c868d347d7d962b43c1f7e6dae9ff4/parser_library/src/workspaces/workspace.cpp#L953) as `proc_grp->suggest(...)` isn't a `const` function

Reply:
I see... that's unfortunate. In that case, I guess we are better off with the original code...